### ISSUE TITLE
[Bug] Touches events with CollectionView on iOS not firing

### ISSUE THREAD

**Issue by josephlbailey** [xamarin/Xamarin.Forms#8401]
### Description

`TouchesBegan`, `TouchesMoved`, `TouchesEnded`, aren't firing when implementing a CollectionView custom renderer. I imagine this probably isn't actually a bug, but has something to do with how the underlying native controls are utilized, but I'll create the issue as if it were a bug just in case.

### Steps to Reproduce

1. Place breakpoints in any of the `Touches*` method overrides in a class implementing `CollectionViewRenderer` that is obviously exported as a custom renderer for a CollectionView.
2. Interact with the CollectionView

### Expected Behavior

Methods should fire when interacting with a CollectionView

### Actual Behavior

Methods not being called

### Basic Information

- Version with issue: 4.3.0947036
- Last known good version: N/A
- IDE: VS Mac (8.3.7)
- Platform Target Frameworks:
  - iOS:  13.2

### Reproduction Link

[CollectionViewTouchBug.zip](https://github.com/xamarin/Xamarin.Forms/files/3815469/CollectionViewTouchBug.zip)

**Comment by hartez** [comment]
This is an issue with the renderer structure. You want to add your gestures on the renderer's `Control` property. 

For example, you can add native tap gestures in `SetupNewElement`:

```
protected override void SetUpNewElement(GroupableItemsView newElement)
{
	base.SetUpNewElement(newElement);

	if (Control == null) 
	{
		return;
	}

	Control.AddGestureRecognizer(new UITapGestureRecognizer(() => System.Diagnostics.Debug.WriteLine($">>>>>> Touched")));
}
```

If you want to override the various `TouchesX` methods, you'll need to subclass UICollectionView and create the appropriate control type in an override of `CreateNativeControl` in your renderer.

Out of curiosity, what were you looking to do by overriding the `TouchesX` methods?

**Comment by josephlbailey** [comment]
@hartez I'm working on a collapsable floating search box at the top of my CollectionView by tapping in to the `scrolled` event, and moving the `Y` position of the search box based on the vertical delta of the scroll. I wanted to get at the `TouchesEnded` event so I could check the position of the search view and if it was > halfway hidden when the touch is released, I would animate the view the rest of the way to its hidden position and vice versa if it was <= halfway hidden. I had a similar issue (#7159) where I created a touch-tracking effect as described in the official documentation [here](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/effects/touch-tracking) and the `DOWN` touch action was not firing in a CollectionView on Android. The recommendation in that issue was for me to create a custom renderer, which I did. Overriding the `DispatchTouchEvent` on Android worked swimmingly. However, on iOS, no such luck. 

I considered using the custom renderer on Android and using the touch effect on iOS, but even the effect was a pretty big dud on iOS because the `TouchesEnded` would only fire if it was a tap and release with no lateral movement. If there was any scrolling, then the `TouchesEnded` would not fire. 

Some days, you just can't win for losing. 😭
